prime-router/src/test/kotlin/fhirengine/utils/FhirTranscoderTests.kt

```diff
@@ -73,7 +73,9 @@ class FhirTranscoderTests {
         val decodedBundle = FhirTranscoder.decode(encodedBundle)
         // Ensure the decoded output is the same as the input pre-encode
         assertThat(decodedBundle).isNotNull()
-        assertThat(decodedBundle).equals(testBundle)
+        assertThat(decodedBundle.id).isEqualTo("Bundle/someid")
+        assertThat(decodedBundle.type.name).isEqualTo(testBundle.type.name)
+        assertThat(decodedBundle.timestamp).isEqualTo(testBundle.timestamp)
     }
```

Comment:
Hmm that is an interesting point. I thought I had gone through and compared the only fields with data, but the testBundle has an entry and the decodedBundle does not. Even if I add data to the entry this happens, is this expected for some reason @carlosfelix2?

Reply:
How were you adding data to the entry? If I add the following after the `testBundle` creation, I do see an entry in both `testBundle` and `decodedBundle`.

```
        val resource = Patient()
        resource.id = "somePatientId"
        testBundle.addEntry().resource = resource
```

Reply:
Awesome, thank you Jimmy. I wonder if it is a bug that it removes an empty entry though. @carlosfelix2 can you speak to this?

Reply:
Added a check to the entry, thank you both for your help on this!

Reply:
> Hmm that is an interesting point. I thought I had gone through and compared the only fields with data, but the testBundle has an entry and the decodedBundle does not. Even if I add data to the entry this happens, is this expected for some reason @carlosfelix2?

This happens because the library is not smart enough and you have to set the resource property like @jimmyfagan said.